### DISCUSSION TITLE
Add right click menu of the filesystem as a button on top of the dock, for web + android usage

### DISCUSSION THREAD

**Discussion by QbieShay** [godotengine/godot-proposals#5733] [Editor]
### Describe the project you are working on

VFX

### Describe the problem or limitation you are having in your project

I am trying to use the web editor from android, and right click doesn't seem to work.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I was blocked at the point where i tried to import a texture - i couldn't find a way to do it

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Not sure :/ sorry

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Maybe? 

### Is there a reason why this should be core and not an add-on in the asset library?

It modifies the editor.

**Comment by Calinou** [comment]
Moved to discussion, as there is no concrete technical implementation provided.

A context menu should be displayed by long-tapping, not with a dedicated button. This is how it works in gallery apps on Android, for instance. This should be implemented in 4.0.beta4 and 3.6.beta already by https://github.com/godotengine/godot/pull/65434 and https://github.com/godotengine/godot/pull/65398 respectively.

However, the Android input refactor has no effect when running on HTML5 – dedicated detection code and input handling code needs to be added for mobile HTML5. Using the native Android editor is still recommended whenever possible as it'll perform better and have better system integration.

> I was blocked at the point where i tried to import a texture - i couldn't find a way to do it

Even if you manage to open a dialog to import files in the web editor, you'll hit another problem anyway. There is no way to access the user's filesystem from the web editor, so using the **Open In File Manager** context menu action won't work. The only way to import files to the web editor is to drag-and-drop them from another window, which is generally not possible to do on Android.
